akka-actor-typed/src/main/scala/akka/actor/typed/internal/ActorContextImpl.scala

```diff
@@ -206,6 +208,14 @@ import akka.util.Timeout
     pipeToSelf((target.ask(createRequest))(responseTimeout, system.scheduler))(mapResponse)
   }
 
+  override def askWithStatus[Req, Res](target: RecipientRef[Req], createRequest: ActorRef[ReplyWithStatus[Res]] => Req)(
+      mapResponse: Try[Res] => T)(implicit responseTimeout: Timeout, classTag: ClassTag[Res]): Unit =
+    ask(target, createRequest) {
+      case Success(ReplyWithStatus.Success(t: Res)) => mapResponse(Success(t))
+      case Success(ReplyWithStatus.Error(why))      => mapResponse(Failure(new RuntimeException(why)))
```

Comment:
we should use a more specific exception type than RuntimeException

Reply:
isn't `why` the exception to use?

Reply:
Indeed, it should not be wrapped here.